Zend/zend_operators.c

```diff
@@ -2540,7 +2540,7 @@ ZEND_API bool ZEND_FASTCALL zend_object_is_true(zval *op) /* {{{ */
 	if (zobj->handlers->cast_object(zobj, &tmp, _IS_BOOL) == SUCCESS) {
 		return Z_TYPE(tmp) == IS_TRUE;
 	}
-	zend_error(E_RECOVERABLE_ERROR, "Object of class %s could not be converted to bool", ZSTR_VAL(zobj->ce->name));
+	zend_type_error("Object of class %s could not be converted to bool", ZSTR_VAL(zobj->ce->name));
```

Comment:
This is generally fine, but we will need to perform an exception-safety audit on all users for zend_is_true/zval_is_true.

Reply:
Not sure how feasible it is do perform this before next Tuesday when RC1 gets tagged as there does seems to be a couple of crucial usages (zend_compare, OpCache) looking at: https://heap.space/search?project=php-src&refs=zval_is_true and https://heap.space/search?project=php-src&refs=zend_is_true

Reply:
I think I've checked all of them but I'm very unsure about how to go about it with some cases.